ui/app/src/components/Blocks/CurrencyForm.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+   <div class="currency-form text--small">
+      <div class="currency-form__input">
+        <SifInput :label="label" />
+      </div>
+      <div class="currency-form__balance">
+        <span>Balance: 1200,7356</span>
+        <SifSelect :tokens="tokens" @change="onChange" />
+      </div>
+    </div>
+</template>
+
+<script>
+import SifButton from '../Elements/SifButton.vue';
+import SifInput from '../Elements/SifInput.vue';
+import SifSelect from '../Elements/SifSelect.vue';
+
+export default {
+  props: {
+    label: {
+      type: String,
+      default: ''
+    }
+  },
+
+  components: {
+    SifButton,
+    SifInput,
+    SifSelect,
+  },
+
+  data() {
+    return {
+      tokens: [
+        {
+          icon: '',
+          name: 'ETH',
+        },
+        {
+          icon: '',
+          name: '0xBTC',
+        },
+        {
+          icon: '',
+          name: 'aDAI',
+        },
+        {
+          icon: '',
+          name: 'AMN',
+        },
+      ]
+    }
+  },
+
+  methods: {
+    onChange(event, data) {
+      console.log(data);
+    }
+  }
+}
+</script>
+
+<style lang="scss">
```

Comment:
Sorry if this is a stupid question from a React guy but Is there any reason we are not using scoped scss? One of the great things you get out of Vue is CSS Modules out of the box so you can avoid having to use strict naming convention systems like SMACSS and BEM and unnecessary nesting.

Reply:
I think the main casualty is readability as you go from this:

```html
<div class="menu">
  <div class="menu__item">
    <a class="menu__link menu__link--active" href="#">Discover</a>
  </div>
  <div class="menu__item">
    <a class="menu__link" href="#">New Wallet</a>
  </div>
  <div class="menu__item">
    <a class="menu__link" href="#">Convert</a>
  </div>
  <div class="menu__item">
    <a class="menu__link" href="#">Deposit</a>
  </div>
</div>
```

```html
<div class="menu">
  <div class="item">
    <a class="link active" href="#">Discover</a>
  </div>
  <div class="item">
    <a class="link" href="#">New Wallet</a>
  </div>
  <div class="item">
    <a class="link" href="#">Convert</a>
  </div>
  <div class="item">
    <a class="link" href="#">Deposit</a>
  </div>
</div>
```

Reply:
Also you might want to pull out the item to its own component and make dingbat__item a block. 

```html
<div class="dingbat__item">
  <a class="dingbat__link" href="#">New Wallet</a>
</div>
```

with styles

```scss
.dingbat {
  ...

  &__item {
    ...
  }

  &__link {
    ...
  }
}
```

This becomes quite different:

```html
<div class="item">
  <a class="item__link" href="#">New Wallet</a>
</div>
```

with styles

```scss
.item {
  ...
  &__link {
      ...
  }
}
```

Where as if we use scoped scss we can generally get away with just cutting and pasting both style and markup.

```html
<div class="item">
  <a class="link" href="#">New Wallet</a>
</div>
```
```scss
.item {
  .link {
    ...
  }
}

```